include/phoenix/script.hh

```diff
@@ -226,6 +227,46 @@ namespace phoenix {
 		const std::type_info* _m_type {nullptr};
 	};
 
+	/// \brief Represents an object associated with an instance in the script.
+	///
+	/// Instances allocated with init_opaque will be backed up by this class with plain memory storage
+	class opaque_instance final : public instance {
+	public:
+		PHOENIX_INTERNAL opaque_instance(symbol const& sym, std::vector<symbol*> const& members);
+		PHOENIX_INTERNAL ~opaque_instance();
+
+		PHOENIX_INTERNAL std::uint8_t* data() {
+			return _m_storage.get();
+		}
+
+		PHOENIX_INTERNAL const std::uint8_t* data() const {
+			return _m_storage.get();
+		}
+
+	private:
+		template <typename T, typename... Args>
+		PHOENIX_INTERNAL T* construct_at(size_t offset, Args&&... args);
+
+		std::unique_ptr<std::uint8_t[]> _m_storage;
+		std::unique_ptr<std::string*[]> _m_strings;
+		size_t _m_str_count {0};
```

Comment:
This feels like a good candidate to use a `std::vector` instead. That would make it easier to understand the intent here.